codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -131,33 +125,34 @@ protected Result beginEncode(HttpResponse httpResponse, String acceptEncoding) t
             return null;
         }
 
-        ZlibWrapper wrapper = determineWrapper(acceptEncoding);
-        if (wrapper == null) {
+        String targetContentEncoding = determineEncoding(acceptEncoding);
+        if (targetContentEncoding == null) {
             return null;
         }
 
-        String targetContentEncoding;
-        switch (wrapper) {
-        case GZIP:
-            targetContentEncoding = "gzip";
-            break;
-        case ZLIB:
-            targetContentEncoding = "deflate";
-            break;
-        default:
+        if (targetContentEncoding.equals("gzip")) {
+            return new Result(targetContentEncoding,
+                    new EmbeddedChannel(ctx.channel().id(), ctx.channel().metadata().hasDisconnect(),
+                            ctx.channel().config(), ZlibCodecFactory.newZlibEncoder(
+                            ZlibWrapper.GZIP, compressionLevel, windowBits, memLevel)));
+        } else if (targetContentEncoding.equals("deflate")) {
+            return new Result(targetContentEncoding,
+                    new EmbeddedChannel(ctx.channel().id(), ctx.channel().metadata().hasDisconnect(),
+                            ctx.channel().config(), ZlibCodecFactory.newZlibEncoder(
+                            ZlibWrapper.ZLIB, compressionLevel, windowBits, memLevel)));
+        } else if (targetContentEncoding.equals("br")) {
+            return new Result(targetContentEncoding,
+                    new EmbeddedChannel(ctx.channel().id(), ctx.channel().metadata().hasDisconnect(),
+                            ctx.channel().config(), new BrotliEncoder()));
+        } else {
             throw new Error();
         }
-
-        return new Result(
-                targetContentEncoding,
-                new EmbeddedChannel(ctx.channel().id(), ctx.channel().metadata().hasDisconnect(),
-                        ctx.channel().config(), ZlibCodecFactory.newZlibEncoder(
-                        wrapper, compressionLevel, windowBits, memLevel)));
     }
 
     @SuppressWarnings("FloatingPointEquality")
-    protected ZlibWrapper determineWrapper(String acceptEncoding) {
+    protected String determineEncoding(String acceptEncoding) {
```

Comment:
We cant do this as its a breaking change

Reply:
Reverted.

Reply:
Even adding this new method does not really help as people that might override the original method will have different behaviour as the old method is not called anymore

Reply:
what you maybe can do is to delegate if its not brotli

Reply:
This is not resolved yet... As you basically break people that override `determineEncoding` as it is not called anymore

Reply:
Do you mean override `determineWrapper` or `determineEncoding`? `determineWrapper` is as it is and not touched. `determineEncoding` is new method which supports Brotli.

I am confused a bit so.

Reply:
@hyperxpro so the problem is that you not call `determineWrapper` anymore with your changes... this means if a user did override it to implement some functionality it will not work anymore as now only `determineEncoding` is coding.

Reply:
Makes sense now. Lemme think of some alternative.

Reply:
@hyperxpro ping me once I should have a look again

Reply:
Sure

Reply:
how about try making a brotli wrapper if the old (untouched) determineWrapper returned null?

Reply:
@valerauko It might never return `null`. If `Accept-Encoding` is `br, gzip, deflate` then `determineWrapper` will always return ZLIB. I am going to extend `HttpContentCompressor` and do Brotli there because I need few parameters too.